lib/puppet/pops/evaluator/evaluator_impl.rb

```diff
@@ -42,19 +42,26 @@ class EvaluatorImpl
   Issues = Issues
 
   def initialize
+    @@initialized ||= static_initialize
+
+    # Use null migration checker unless given in context
+    @migration_checker = (Puppet.lookup(:migration_checker) { Migration::MigrationChecker.singleton() })
```

Comment:
Two pairs of unnecessary parenthesis here. One around the expression and one after singleton.